src/gui/transferlistwidget.cpp

```diff
@@ -807,6 +812,38 @@ void TransferListWidget::askAddTagsForSelection()
         addSelectionTag(tag);
 }
 
+void TransferListWidget::editTorrentTrackers()
+{
+    const QVector<BitTorrent::TorrentHandle *> torrents = getSelectedTorrents();
+    QVector<BitTorrent::TrackerEntry> commonTrackers;
+
+    if (!torrents.empty()) {
+        commonTrackers = torrents[0]->trackers();
+
+        for (const BitTorrent::TorrentHandle *torrent : torrents) {
+            QSet<BitTorrent::TrackerEntry> trackerSet;
+
+            for (const BitTorrent::TrackerEntry &entry : asConst(torrent->trackers()))
+                trackerSet.insert(entry);
+
+            commonTrackers.erase(std::remove_if(commonTrackers.begin(), commonTrackers.end()
+                , [&trackerSet](const BitTorrent::TrackerEntry &entry) { return !trackerSet.contains(entry); })
+                , commonTrackers.end());
+        }
+    }
+
+    auto trackerDialog = new TrackerEntriesDialog(this);
+    trackerDialog->setAttribute(Qt::WA_DeleteOnClose);
+    trackerDialog->setTrackers(commonTrackers);
+
+    connect(trackerDialog, &QDialog::accepted, this, [torrents, trackerDialog]() {
+        for (BitTorrent::TorrentHandle *torrent : torrents)
+            torrent->replaceTrackers(trackerDialog->trackers());
```

Comment:
Dialog allow you to have several trackers with the same URL (but with the different or even the same tier). What happens when you add such a list to torrent?

Reply:
The list is accepted as is. I think user is responsible for such action, it might even be user's intention.

Reply:
>The list is accepted as is. I think user is responsible for such action, it might even be user's intention.

I mean how the duplicates are handled by libtorrent?
>However the Trackers tab in GUI doesn't show the full list (duplicate trackers), I consider it a bug and should be fixed in another PR.

Maybe libtorrent just filters out duplicates?

Reply:
>I mean how the duplicates are handled by libtorrent?

Same, it is accepted as-is.
However it only preserves the lowest tier of duplicated trackers when writing to fastresume, so after qbt restarts it will give the trimmed list.

Reply:
Correction: the trackers are written to fastresume as is, but `m_nativeHandle.trackers()` seems to only return the lowest tier of duplicated trackers.

Reply:
OK, let's leave it to libtorrent.

Reply:
> Correction: the trackers are written to fastresume as is, but `m_nativeHandle.trackers()` seems to only return the lowest tier of duplicated trackers.

Actually they are filtered out when loading fastresume data.
Also there are some more inconsistensies in libtorrent code. See https://github.com/arvidn/libtorrent/issues/6846.